Build failure: qtwayland 5.15.14

### Steps To Reproduce

Steps to reproduce the behavior:
1. Run `nix run ./#build-switch` (from this template), triggering nix-darwin to be rebuilt
2. Build qtwayland (I assume as part of darwin).

### Build log

```
❯ nix run ./#build-switch
warning: Git tree '/Users/foobar/nixos-config' is dirty
Running build-switch for aarch64-darwin
Starting build...
warning: Git tree '/Users/foobar/nixos-config' is dirty
error: builder for '/nix/store/plnyry7c34inljgngnnlv2spw08s5a0b-qtwayland-5.15.14.drv' failed with exit code 2;
       last 10 log lines:
       > mv -f libQt5WaylandCompositor.5.15.14.dylib ../../lib/libQt5WaylandCompositor.5.15.14.dylib
       > rm -f ../../lib/libQt5WaylandCompositor.dylib
       > rm -f ../../lib/libQt5WaylandCompositor.5.dylib
       > rm -f ../../lib/libQt5WaylandCompositor.5.15.dylib
       > mv -f libQt5WaylandCompositor.dylib ../../lib/libQt5WaylandCompositor.dylib
       > mv -f libQt5WaylandCompositor.5.dylib ../../lib/libQt5WaylandCompositor.5.dylib
       > mv -f libQt5WaylandCompositor.5.15.dylib ../../lib/libQt5WaylandCompositor.5.15.dylib
       > make[2]: Leaving directory '/private/tmp/nix-build-qtwayland-5.15.14.drv-0/qtwayland-75f434f/src/compositor'
       > make[1]: Leaving directory '/private/tmp/nix-build-qtwayland-5.15.14.drv-0/qtwayland-75f434f/src'
       > make: *** [Makefile:52: sub-src-make_first] Error 2
       For full logs, run 'nix log /nix/store/plnyry7c34inljgngnnlv2spw08s5a0b-qtwayland-5.15.14.drv'.
error: 1 dependencies of derivation '/nix/store/z3g3j6la9gn69gjacc061caxp2m4dals-urh-2.9.6.drv' failed to build
error: 1 dependencies of derivation '/nix/store/4k030gmsbsaw3j1m0gz4kz465zs20al4-home-manager-applications.drv' failed to build
error: 1 dependencies of derivation '/nix/store/s54284619cgq3wjw17aalqklr1qyg560-home-manager-fonts.drv' failed to build
error: 1 dependencies of derivation '/nix/store/071hm4dpgrr9qicdjrpzsp99m58vfai3-home-manager-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/n6y3z3jxa79pyc2jbq4hq663qls0i8sf-man-paths.drv' failed to build
error: 1 dependencies of derivation '/nix/store/hc0xhxbkhj8x501fyasai6iqxmf2qk0r-system-applications.drv' failed to build
error: 1 dependencies of derivation '/nix/store/ndmajrgjpjcnpq2ypfm95mgxyi2nyx75-urh-2.9.6-fish-completions.drv' failed to build
error: 1 dependencies of derivation '/nix/store/py0d5ds7ngl5qi1qclzxhl93wj8l374z-darwin-system-24.11.20240714.693bc46+darwin4.5ce8503.drv' failed to build
```

Here is the output of `nix log /nix/store/plnyry7c34inljgngnnlv2spw08s5a0b-qtwayland-5.15.14.drv`:

https://gist.github.com/victorhooi/de229a867862541df4270ea8ae19a1d2

### Additional context

Add any other context about the problem here.

### Notify maintainers

@qknight @ttuegel @periklis @bkchr @wegank @NickCao @alyssais @OPNA2608 @K900 

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
❯ nix-shell -p nix-info --run "nix-info -m"
error:
       … while calling the 'import' builtin
         at «string»:1:18:
            1| {...}@args: with import <nixpkgs> args; (pkgs.runCommandCC or pkgs.runCommand) "shell" { buildInputs = [ (nix-info) ]; } ""
             |                  ^

       … while realising the context of a path

       … while calling the 'findFile' builtin
         at «string»:1:25:
            1| {...}@args: with import <nixpkgs> args; (pkgs.runCommandCC or pkgs.runCommand) "shell" { buildInputs = [ (nix-info) ]; } ""
             |                         ^

       error: file 'nixpkgs' was not found in the Nix search path (add it using $NIX_PATH or -I)

nixos-config on  main [!] 
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/radio/urh/default.nix
+++ b/pkgs/applications/radio/urh/default.nix
@@ -1,4 +1,4 @@
-{ lib, fetchFromGitHub, python3Packages
+{ stdenv, lib, fetchFromGitHub, python3Packages
 , hackrf, rtl-sdr, airspy, limesuite, libiio
 , libbladeRF
 , qt5
@@ -16,8 +16,9 @@ python3Packages.buildPythonApplication rec {
   };
 
   nativeBuildInputs = [ qt5.wrapQtAppsHook ];
-  buildInputs = [ hackrf rtl-sdr airspy limesuite libiio libbladeRF qt5.qtwayland ]
-    ++ lib.optional USRPSupport uhd;
+  buildInputs = [ hackrf rtl-sdr airspy limesuite libiio libbladeRF ]
+    ++ lib.optional USRPSupport uhd
+    ++ lib.optional stdenv.hostPlatform.isLinux qt5.qtwayland;
 
   propagatedBuildInputs = with python3Packages; [
     pyqt5 numpy psutil cython pyzmq pyaudio setuptools

